src/coreclr/jit/ssabuilder.cpp

```diff
@@ -799,6 +808,11 @@ void SsaBuilder::RenameDef(GenTreeOp* asgNode, BasicBlock* block)
 
         lclNode->SetSsaNum(SsaConfig::RESERVED_SSA_NUM);
     }
+    else if (defNode->OperIs(GT_CALL))
+    {
+        // GT_CALL that doesn't define local, nothing to do further.
```

Comment:
I don't think it is correct to omit the memory renaming here.

Consider an otherwise pure call that defined a struct which has been address-exposed. We will miss pushing a definition for memory in that case.

I don't think we have calls right now that return structs via return buffer and are pure like that, but it does not seem right to bake that assumption into this code (especially without asserts).

Reply:
Agree, I think I meant to remove this but forgot.

Reply:
Let me know if I missed anything.

Reply:
I see what you mean, and agree the code is correct.

It is a little tricky though, the assumption here is that `!isLocal` calls always induce (GC heap) havoc. If we delete the condition, we drop that assumption, but are still otherwise always correct, so I think it is still better to delete it.

Reply:
I was hitting this assert if we remove this condition. 

https://github.com/dotnet/runtime/blob/00ed84ae9ba5c68ada6041ca4aa8083cc89e0669/src/coreclr/jit/valuenum.cpp#L7192

Basically, we were wrongly adding the def and hence PHI for a block, but the PHI just had 1 argument. The repro didn't even had any calls with hidden buffer. By having this check we at least guarantee that the existing behavior without return buffer calls won't be affected.